### ISSUE TITLE
run-commands cuts last line of inquirer output/console output

### ISSUE THREAD

**Issue by BlumDev92** [nrwl/nx#13540]
### Current Behavior

I'm running a custom node script for my executor `nx run my-app:my-custom-script`. My script contains an inquirer form. NX cuts the last line of this form.

I'm using:
```
"inquirer": "^9.1.4"
```

Sample code of inquirer form (`index.mjs`):
```
[...]
  const questions = [
    {
      type: 'list',
      name: 'SAMPLE',
      message: 'This is a samle question?',
      choices: ['Option1', 'Option2', 'Option3', 'Option4'],
    }
  ];

  const answers = await inquirer.prompt(questions);
[...]
```

`project.json`:
```
[...]
"my-custom-script": {
      "executor": "nx:run-commands",
      "options": {
        "commands": [
          {
            "command": "node build/index.mjs"
          }
        ],
        "cwd": "apps/my-app",
        "parallel": false
      }
    }
[...]
```

Sample output when I run `nx run my-app:my-custom-script`:
```
? This is a samle question? (Use arrow keys)
❯ Option1 
  Option2 
  Option3 

```

When I press `CTRL+C` the missing output gets printed:
```
? This is a samle question? (Use arrow keys)
❯ Option1 
  Option2 
  Option3 
  Option4  <--- PRINTED!
 >  NX   ERROR: Something went wrong in run-commands - Command failed: node build/index.mjs
   Pass --verbose to see the stacktrace.
```

### Expected Behavior

Console output gets printed with last row.

Expected console output:
```
? This is a samle question? (Use arrow keys)
❯ Option1 
  Option2 
  Option3 
  Option4
```

### Steps to Reproduce

1. Create a node script and put it somewhere in workspace.
2. Create target like above in project.json of any app.
3. Run new created target

### Nx Report

```shell
Node : 16.13.0
   OS   : darwin x64
   npm  : 8.1.0
   
   nx : 15.2.1
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.2.1
   @nrwl/detox : 15.2.1
   @nrwl/devkit : 15.2.1
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.2.1
   @nrwl/expo : 15.2.1
   @nrwl/express : Not Found
   @nrwl/jest : 15.2.1
   @nrwl/js : 15.2.1
   @nrwl/linter : 15.2.1
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.2.1
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.2.1
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.2.1
   @nrwl/web : 15.2.1
   @nrwl/webpack : 15.2.1
   @nrwl/workspace : 15.2.1
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

**Comment by jeffhertzler** [comment]
this happens when using `nx exec` as well

**Comment by MohammadKurjieh** [comment]
Any updates on this issue? I am also facing it nx version 15.8.6.

**Comment by enchorb** [comment]
+1

**Comment by Lepozepo** [comment]
Very annoying problem to have ><

**Comment by abrahamsaanchez** [comment]
+1

I can't even select an option.

**Comment by Lepozepo** [comment]
It seems this issue also affects other logged outputs as well, more importantly the last line of running tests interactively on something like jest

**Comment by Schmale97** [comment]
+1

I am also experiencing this when using inquirer inside a local executor

**Comment by Lepozepo** [comment]
I'm seeing the same behavior on most (if not all) other logs as well outside of `inquirer`

**Comment by abelpz** [comment]
any updates on this? The same thing happens to me.

**Comment by samrho** [comment]
There's a workaround for this.
Instead of the last option being cut off, the separator you will add at the end of the list will be cut off.

```typescript
import inquirer from 'inquirer';

const questions = [
  {
    type: 'list',
    name: 'SAMPLE',
    message: 'This is a samle question?',
    choices: [
      'Option1',
      'Option2',
      'Option3',
      'Option4',
      new inquirer.Separator() // <- add a seperator.
    ],
  }
];

const answers = await inquirer.prompt(questions);
```

**Comment by pietschy** [comment]
I'm seeing this as well - tried swapping to enquirer but it's behaving the same.  I'm using nx@16.9.0

**Comment by mcfdez** [comment]
I confirm that with the version nx@16.9.0 I reproduce the error.

**Comment by GreedyA1** [comment]
this happens to me in custom executor. the same function works fine in custom generator 🤔

**Comment by AgentEnder** [comment]
This should be fixed with the native comand runner in Nx 17.3. I'm going to go ahead with closing this out as the associated work has been merged.
